src/groupByKey.ts

```diff
@@ -1,7 +1,30 @@
+/* eslint-disable no-restricted-syntax */
 type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const arrOfValues: string[] | number[] = [];
+  const result: GroupsMap<T> = {};
+
+  // eslint-disable-next-line no-restricted-syntax
+  for (const item of items) {
+    if (!arrOfValues.includes(item[key])) {
+      arrOfValues.push(item[key]);
+    }
+  }
```

Comment:
```suggestion

```
Also, you can simplify the logic a bit, work with a result, and use just one cycle here